htsget-config/examples/config-files/c4gh.toml

```diff
@@ -1,23 +1,19 @@
 # An example of running htsget-rs with Crypt4GH enabled.
-# Run with `cargo run -p htsget-axum --features experimental -- --config htsget-config/examples/config-files/c4gh.toml`
+# Run with `cargo run -p htsget-axum --features experimental,s3-storage -- --config htsget-config/examples/config-files/c4gh.toml`
```

Comment:
`s3-storage` shouldn't be required, IMHO?

Reply:
Yeah that's fair. It's required because the secrets manager has all the AWS dependencies, which `s3-storage` enables.

Originally, only S3 was required for any AWS-specific things, but now that's changed, maybe this feature should be called `aws` rather than `s3-storage`.

Reply:
It might be good to rename/simplify `url-storage` too? E.g.

`s3-storage` -> `aws`
`url-storage` -> `url`

Reply:
I'll make this a separate PR.

Reply:
Agreed on the renaming, but shouldn't we have a `file://` for the secrets management part? I'm thinking about local testing and/or devving, right?

Reply:
There is `kind = "File"` for local keys:

```
backend.keys.kind = "File"
backend.keys.private = "data/c4gh/keys/bob.sec" # pragma: allowlist secret
backend.keys.public = "data/c4gh/keys/alice.pub"
```

And `kind = "SecretsManager"` for secrets manager keys:
```
backend.keys.kind = "SecretsManager"
backend.keys.private = "htsget/test_c4gh_private_key" # pragma: allowlist secret
backend.keys.public = "htsget/test_c4gh_public_key"
```

Is that what you mean? Or some sort of local SecretsManager option?

Reply:
No, I'm only concerned about being able to run c4gh infra locally, without requiring `aws` deps... that "Run with `cargo run...`" line change seems to imply that `s3-storage` is absolutely required?

Reply:
Yes, you're right, it's not required. It only required if you remove the comments in that file that use `kind = "SecretsManager"`.

I'll make this clearer on the next PR. Maybe it should just be `--all-features` in all the examples to avoid any confusion?

Reply:
> Maybe it should just be --all-features in all the examples to avoid any confusion?

🤔 ... that'd exercise our error management logic and corresponding error messages, i.e: running with `file://` paths and `aws`-related c4gh secrets management. It might make more sense this way if errors are clear, yeah!